datafusion/expr/src/expr.rs

```diff
@@ -3818,7 +3816,7 @@ mod test {
         // If this test fails when you change `Expr`, please try
         // `Box`ing the fields to make `Expr` smaller
         // See https://github.com/apache/datafusion/issues/16199 for details
-        assert_eq!(size_of::<Expr>(), 128);
+        assert_eq!(size_of::<Expr>(), 112);
```

Comment:
This change saves 16 bytes per Expr. Nice. 

I will run some planning benchmarks and see if we can see any difference